File: CS-Examples/02_Data/FormatTable.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;
using Spire.Xls.Core.Spreadsheet.PivotTables;

namespace FormatTable
{
    public partial class Form1 :Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, System.EventArgs e)
        {
            // Create a new workbook instance
            Workbook workbook = new Workbook();

            // Load an Excel file into the workbook
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\FormatTable.xlsx");

            // Get the first worksheet from the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Add a default table style to the table in the worksheet
            sheet.ListObjects[0].BuiltInTableStyle = TableBuiltInStyles.TableStyleMedium9;
            
            // Show total row for the table
            sheet.ListObjects[0].DisplayTotalRow = true;

            //Set calculation type
            sheet.ListObjects[0].Columns[0].TotalsRowLabel = "Total";
            sheet.ListObjects[0].Columns[1].TotalsCalculation = ExcelTotalsCalculation.None;
            sheet.ListObjects[0].Columns[2].TotalsCalculation = ExcelTotalsCalculation.None;
            sheet.ListObjects[0].Columns[3].TotalsCalculation = ExcelTotalsCalculation.Sum;
            sheet.ListObjects[0].Columns[4].TotalsCalculation = ExcelTotalsCalculation.Sum;

            // Show row stripes and column stripes using table style
            sheet.ListObjects[0].ShowTableStyleRowStripes = true;
            sheet.ListObjects[0].ShowTableStyleColumnStripes = true;

            // Save the modified workbook to a new file named "Sample.xlsx" using Excel 2010 version
            workbook.SaveToFile("Sample.xlsx", ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the file
            ExcelDocViewer("Sample.xlsx");
        }

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: CS-Examples/02_Data/FormatTable.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet.PivotTables

Namespace FormatTable
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()
            ' Load an existing Excel file 
            workbook.LoadFromFile("..\..\..\..\..\..\Data\FormatTable.xlsx")
            ' Get the first worksheet from the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Add Default Style to the table
            sheet.ListObjects(0).BuiltInTableStyle = TableBuiltInStyles.TableStyleMedium9

            ' Show Total
            sheet.ListObjects(0).DisplayTotalRow = True

            ' Column 1 will display "Total" in the totals row
            sheet.ListObjects(0).Columns(0).TotalsRowLabel = "Total"
            ' Column 2 will have no totals calculation
            sheet.ListObjects(0).Columns(1).TotalsCalculation = ExcelTotalsCalculation.None
            ' Column 3 will have no totals calculation
            sheet.ListObjects(0).Columns(2).TotalsCalculation = ExcelTotalsCalculation.None
            ' Column 4 will calculate the sum
            sheet.ListObjects(0).Columns(3).TotalsCalculation = ExcelTotalsCalculation.Sum
            ' Column 5 will calculate the sum
            sheet.ListObjects(0).Columns(4).TotalsCalculation = ExcelTotalsCalculation.Sum
            ' Enable row stripes in the table style
            sheet.ListObjects(0).ShowTableStyleRowStripes = True
            ' Enable column stripes in the table style
            sheet.ListObjects(0).ShowTableStyleColumnStripes = True
            ' Save the modified workbook to a new file named "Sample.xlsx" in Excel 2010 format
            workbook.SaveToFile("Sample.xlsx", ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()
            ExcelDocViewer("Sample.xlsx")
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
